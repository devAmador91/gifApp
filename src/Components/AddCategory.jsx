import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState("");


  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setInputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form aria-label="form" onSubmit={(e) => onSubmit(e)}>
      <input
        onChange={(e) => handleChange(e)}
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
