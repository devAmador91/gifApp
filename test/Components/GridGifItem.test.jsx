import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/Components/GifGridItem";

describe("Prueba en <GridGifItem/>", () => {
  const title = "Pokemon 2000";
  const url = "https://media.tenor.com/R1KctPuHA8sAAAAd/miedo-perro.gif";
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifGridItem title={title} url={url} />);
    //--> Sirve para visualizar el HTML screen.debug();

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe mostrar el titulo en el componente', () => {

    render(<GifGridItem title={title} url={url}/>);
    expect(screen.getByText(title)).toBeTruthy();

   })
});
