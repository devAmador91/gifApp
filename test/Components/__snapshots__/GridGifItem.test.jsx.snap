// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GridGifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon 2000"
      src="https://media.tenor.com/R1KctPuHA8sAAAAd/miedo-perro.gif"
    />
    <p>
      Pokemon 2000
    </p>
  </div>
</div>
`;
